opm/simulators/wells/StandardWellConnections.cpp

```diff
@@ -95,212 +207,257 @@ computeDensities(const std::vector<Scalar>& perfComponentRates,
-                if (perf == 0) { //
-                    switch (well_.wellEcl().getPreferredPhase()) {
-                    case Phase::OIL:
-                        mix[FluidSystem::oilCompIdx] = 1.0;
-                        break;
-                    case Phase::GAS:
-                        mix[FluidSystem::gasCompIdx] = 1.0;
-                        break;
-                    case Phase::WATER:
-                        mix[FluidSystem::waterCompIdx] = 1.0;
-                        break;
-                    default:
-                        // No others supported.
-                        break;
-                    }
-                // For the rest of the perforation without flow we use mix from the above perforation.
-                } else {
-                    mix = x;
-                }
-
-            }
-        }
-        // Compute volume ratio.
-        x = mix;
-
-        // Subtract dissolved gas from oil phase and vapporized oil from gas phase and vaporized water from gas phase
-        if (FluidSystem::phaseIsActive(FluidSystem::gasPhaseIdx) && FluidSystem::phaseIsActive(FluidSystem::oilPhaseIdx)) {
-            const unsigned gaspos = Indices::canonicalToActiveComponentIndex(FluidSystem::gasCompIdx);
-            const unsigned oilpos = Indices::canonicalToActiveComponentIndex(FluidSystem::oilCompIdx);
-            Scalar rs = 0.0;
-            Scalar rv = 0.0;
-            if (!props.rsmax_perf.empty() && mix[oilpos] > 1e-12) {
-                rs = std::min(mix[gaspos] / mix[oilpos], props.rsmax_perf[perf]);
             }
-            if (!props.rvmax_perf.empty() && mix[gaspos] > 1e-12) {
-                rv = std::min(mix[oilpos] / mix[gaspos], props.rvmax_perf[perf]);
-            }
-            const Scalar d = 1.0 - rs*rv;
-            if (d <= 0.0) {
-                std::ostringstream sstr;
-                sstr << "Problematic d value " << d << " obtained for well " << well_.name()
-                     << " during computeConnectionDensities with rs " << rs
-                     << ", rv " << rv
-                     << " obtaining d " << d
-                     << " Continue as if no dissolution (rs = 0) and vaporization (rv = 0) "
-                     << " for this connection.";
-                deferred_logger.debug(sstr.str());
-            } else {
-                if (rs > 0.0) {
-                    // Subtract gas in oil from gas mixture
-                    x[gaspos] = (mix[gaspos] - mix[oilpos]*rs)/d;
-                }
-                if (rv > 0.0) {
-                    // Subtract oil in gas from oil mixture
-                    x[oilpos] = (mix[oilpos] - mix[gaspos]*rv)/d;
-                }
+            else {
+                // For the rest of the perforations without flow we use the
+                // previousMixtureture from the perforation above.
+                previousMixture = currentMixture;
```

Comment:
I do not understand the previous/current naming. The original names mix and x were not good, but I do not see any relation to time or changing state here that the terms previous or current imply. What about surfVolMix and volMix for example, maybe that would be clearer?

Reply:
> I do not understand the previous/current naming

Current and previous were meant to refer to the the current and previous perforations in the index sequence `0..nperf-1`.  That said, I also wasn't particularly happy with that naming convention so I've pushed an update that uses `componentMixture` and `phaseMixture` instead.